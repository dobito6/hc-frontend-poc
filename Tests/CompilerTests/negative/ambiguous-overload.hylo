//! no-std

struct A {}
struct B {}

fun f(x: A) {}
fun f(x: B) {}

fun g() {
  f(x: ())
}
