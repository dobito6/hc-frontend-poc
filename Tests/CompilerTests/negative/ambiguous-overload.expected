negative/ambiguous-overload.hylo:8.3-4: error: ambiguous use of 'f'
  f(x: ())
  ^
negative/ambiguous-overload.hylo:8.8-10: error: cannot pass value of type 'Void' to parameter 'let A'
  f(x: ())
       ~~
